packages/core/src/transaction.ts

```diff
@@ -603,3 +633,20 @@ export function assertTransactionIsCompatibleWithOptions(transaction: Transactio
     );
   }
 }
+
+function assertSupportedTransactionOptions(
+  sequelize: Sequelize,
+  options: TransactionOptions | NormalizedTransactionOptions,
+) {
+  if (options.readOnly && !sequelize.dialect.supports.startTransaction.readOnly) {
```

Comment:
isn't this also used to determine whether it will access the read pool instead of the write pool when replication is active? This option should be allowed, just not passed to `startTransaction` if not supported

Reply:
I've had a quick search of the code base for `readOnly` and the only use of it I could find was in transactions.

I'm happy to remove it if this is not the case but I couldn't find any other references to the option.

Reply:
It is only used in transactions, but it's used to make the transaction's call to `connectionManager.getConnection` pick a read replica in all dialects instead of the primary, not just those that support `START TRANSACTION READONLY`

Reply:
Got it, I've updated this now